src/kbmod_wf/task_impls/reproject_wu.py

```diff
@@ -12,3 +23,289 @@ def reproject_wu(input_wu=None, reprojected_wu=None, logger=None):
+
+
+def reproject_wu(
+    original_wu_filepath: str = None,
+    uri_filepath: str = None,
+    reprojected_wu_filepath: str = None,
+    runtime_config: dict = {},
+    logger: Logger = None,
+):
+    wu_reprojector = WUReprojector(
+        original_wu_filepath=original_wu_filepath,
+        uri_filepath=uri_filepath,
+        reprojected_wu_filepath=reprojected_wu_filepath,
+        runtime_config=runtime_config,
+        logger=logger,
+    )
+
+    return wu_reprojector.reproject_workunit()
+
+
+class WUReprojector:
+    def __init__(
+        self,
+        original_wu_filepath: str = None,
+        uri_filepath: str = None,
+        reprojected_wu_filepath: str = None,
+        runtime_config: dict = {},
+        logger: Logger = None,
+    ):
+        self.original_wu_filepath = original_wu_filepath
+        self.uri_filepath = uri_filepath
+        self.reprojected_wu_filepath = reprojected_wu_filepath
+        self.runtime_config = runtime_config
+        self.logger = logger
+
+        self.overwrite = self.runtime_config.get("overwrite", False)
+        self.search_config = self.runtime_config.get("search_config", None)
+
+        # Default to 8 workers if not in the config. Value must be 0<num workers<65.
+        self.n_workers = np.max(1, np.min(self.runtime_config.get("n_workers", 8), 64))
+
+        # ! Decide if we want these from a runtime_config or exclusively read from the uri file
+        self.patch_corners = self.runtime_config.get("patch_corners", None)
+        self.guess_dist = self.runtime_config.get("guess_dist", None)
+        self.pixel_scale = self.runtime_config.get("pixel_scale", None)
+
+        # ! Decide if we want these from a runtime_config or calculate each time based on self.uri_params
+        self.image_width = self.runtime_config.get("image_width", None)
+        self.image_height = self.runtime_config.get("image_height", None)
+
+        self.uri_params = self._get_params_from_uri_file(uri_file=self.uri_file)
+
+        self.pixel_scale = self.uri_params["pixel_scale"]
+        self.patch_size = self.uri_params["patch_size"]
+
+        # handle image dimensions
+        if self.image_width == None or self.image_height == None:
+            if "patch_size" not in self.uri_params:
+                raise KeyError(
+                    f"Must supply image dimensions (image_width, image_height) or #patch_size= must be in a specified URI file."
+                )
+            if self.pixel_scale == None:
+                raise KeyError(
+                    f"When patch pixel dimensions are not specifified, the user must supply a pixel scale via the command line or the uri file."
+                )
+
+        self.image_width, self.image_height = self._patch_arcmin_to_pixels(
+            patch_size_arcmin=self.patch_size,
+            pixel_scale_arcsec_per_pix=self.pixel_scale,
+        )
+
+        self.point_on_earth = EarthLocation(1814303.74553723, -5214365.7436216, -3187340.56598756, unit="m")
```

Comment:
@wilsonbb I think this is an odd thing to hard code. Do you have thoughts on how we could parameterize this and stash it in the runtime_config perhaps?

Reply:
I agree, we can put it in some sort of config and the proper way should instead be`EarthLocation.of_site("ctio")` similar to [here](https://github.com/dirac-institute/kbmod/blob/f160686227bba4bfa4818f0e47b5fb2d3188f515/src/kbmod/work_unit.py#L891) 

So ideally we should be making the observatory location name (in this case "ctio") configurable

Reply:
Perfect, updated this.